lib/s3middleware/azureHelpers/mpuUtils.js

```diff
@@ -172,18 +172,21 @@ dataStoreName, log, cb) => {
                 `Error returned from Azure: ${err}`));
         }
         const numberSubParts = results.length;
+        if (numberSubParts !== subPartInfo.expectedNumberSubParts) {
+            return cb(errors.InternalError.customizeDescription('Unexpected ' +
+            'result streaming part to Azure. Please retry.'));
+        }
```

Comment:
what is left in azure at this point? will a new put overwrite or are we going to have a size issue?

Reply:
A new put will overwrite; but if somehow we had streamed one more subpart than expected, that block will remain and take up one block of the total block limit. It shouldn't happen anyway, but it's a safety check. Even if it does happen, we would still be able to retrieve only the correct data.

Reply:
@LaurenSpiegel does returning an InternalError make sense here or still return ServiceUnavailable?

Reply:
should be an assertion since it is a code problem.

Reply:
updated